File: cpp/0136只出现一次的数字.cpp
#include<vector>
#include<iostream>
#include<algorithm>

using namespace std;

int singleNumber1(vector<int> &nums)
{
    int res = 0;
    for(int i :nums)
        res ^= i;
    return res;
}

int singleNumber2(vector<int> &nums)
{
    sort(nums.begin(),nums.end());
    int i = 0;
    for (; i < nums.size() - 1; ++i)
    {
        if(nums[i]!=nums[i+1])
            return nums[i];
        ++i;
    }
    return nums[i];
}

File: python/main/0136只出现一次的数字.py
a=[4,1,2,1,2]


def singleNumber1(nums:list):
    d=dict()
    for i in nums:
        if(i not in d.keys()):
            d[i]=1
        else:
            d[i]+=1
    
    for i in d.keys():
        if(d[i]==1):
            return i

def singleNumber2(nums:list):
    s=set(nums)
    return sum(s)*2-sum(nums)

# 异或的交换律
def singleNumber3(nums:list):
    res=0
    for i in nums:
        res^=i
    
    return res

